libkbfs/root_metadata.go

```diff
@@ -795,14 +823,39 @@ func (rmds *RootMetadataSigned) Version() MetadataVer {
 	// new version.
 	if len(rmds.MD.Extra.UnresolvedWriters) > 0 ||
 		len(rmds.MD.UnresolvedReaders) > 0 ||
-		rmds.MD.ConflictInfo != nil {
+		rmds.MD.ConflictInfo != nil ||
+		rmds.MD.FinalizedInfo != nil {
 		return InitialExtraMetadataVer
 	}
 	// Let other types of MD objects use the older version since they
 	// are still compatible with older clients.
 	return PreExtraMetadataVer
 }
 
+// MakeFinalCopy returns a complete copy of this RootMetadataSigned (but with
+// cleared serialized metadata), with the revision incremented and the final bit set.
+func (rmds *RootMetadataSigned) MakeFinalCopy(config Config) (
+	*RootMetadataSigned, error) {
+	if rmds.MD.IsFinal() {
+		return nil, MetadataIsFinalError{}
+	}
+	var newRmds RootMetadataSigned
+	if err := CodecUpdate(config.Codec(), &newRmds, rmds); err != nil {
+		return nil, err
+	}
+	err := rmds.MD.deepCopyInPlace(config.Codec(), false, &newRmds.MD)
+	if err != nil {
+		return nil, err
+	}
+	// clear the serialized data.
+	newRmds.MD.SerializedPrivateMetadata = nil
+	// set the final flag
+	newRmds.MD.Flags |= MetadataFlagFinal
+	// increment revision but keep the prev root
```

Comment:
What's the reasoning behind keeping the prev root field the same?  Isn't it still useful to have a complete linear chain, even if the field is not properly signed over?  Since this PR doesn't include the `MDOps` changes needed to consume this finalized MD, it's hard for me to tell how it will be used.